File: src/commonTest/kotlin/com/scoreo/application/ArchiveGameTypeUseCaseTest.kt
package com.scoreo.application

import com.scoreo.infrastructure.InMemoryGameTypeRepository
import com.scoreo.domain.model.GameType
import com.scoreo.domain.model.WinCondition
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

class ArchiveGameTypeUseCaseTest {

    private lateinit var gameTypeRepository: InMemoryGameTypeRepository
    private lateinit var useCase: ArchiveGameTypeUseCase

    @Test
    fun `archive game type sets active to false`() {
        gameTypeRepository = InMemoryGameTypeRepository()
        useCase = ArchiveGameTypeUseCase(gameTypeRepository)
        
        val gameType = GameType(id = "g1", name = "Belote", winCondition = WinCondition.HIGHEST_SCORE)
        gameTypeRepository.save(gameType)
        
        useCase.invoke("g1")
        
        val archived = gameTypeRepository.findById("g1")!!
        assertEquals(false, archived.active)
    }

    @Test
    fun `archive preserves game type id and name`() {
        gameTypeRepository = InMemoryGameTypeRepository()
        useCase = ArchiveGameTypeUseCase(gameTypeRepository)
        
        val gameType = GameType(id = "g1", name = "Belote", winCondition = WinCondition.HIGHEST_SCORE)
        gameTypeRepository.save(gameType)
        
        useCase.invoke("g1")
        
        val archived = gameTypeRepository.findById("g1")!!
        assertEquals("g1", archived.id)
        assertEquals("Belote", archived.name)
    }

    @Test
    fun `archive non-existent game type throws`() {
        gameTypeRepository = InMemoryGameTypeRepository()
        useCase = ArchiveGameTypeUseCase(gameTypeRepository)
        
        assertFailsWith<IllegalArgumentException> {
            useCase.invoke("nonexistent")
        }
    }
}

File: apps/scoreo/src/application/archiveGameTypeUseCase.test.ts
import { describe, expect, it } from 'vitest'
import { NotFoundError } from '../domain/model/errors'
import { InMemoryGameTypeRepository } from '../infrastructure/testing/inMemoryGameTypeRepository'
import { ArchiveGameTypeUseCase } from './archiveGameTypeUseCase'

describe('ArchiveGameTypeUseCase', () => {
  it('archive game type sets active to false', () => {
    const repo = new InMemoryGameTypeRepository()
    const useCase = new ArchiveGameTypeUseCase(repo)

    repo.save({
      id: 'g1',
      name: 'Belote',
      winCondition: 'HIGHEST_SCORE',
      tieBreakRule: 'NONE',
      tieBreakCondition: 'HIGHEST_SCORE',
      tieBreakLabel: null,
      moduleId: null,
      active: true,
    })

    useCase.invoke('g1')

    expect(repo.findById('g1')?.active).toBe(false)
  })

  it('archive preserves game type id and name', () => {
    const repo = new InMemoryGameTypeRepository()
    const useCase = new ArchiveGameTypeUseCase(repo)

    repo.save({
      id: 'g1',
      name: 'Belote',
      winCondition: 'HIGHEST_SCORE',
      tieBreakRule: 'NONE',
      tieBreakCondition: 'HIGHEST_SCORE',
      tieBreakLabel: null,
      moduleId: null,
      active: true,
    })

    useCase.invoke('g1')

    const archived = repo.findById('g1')
    expect(archived?.id).toBe('g1')
    expect(archived?.name).toBe('Belote')
  })

  it('archive non-existent game type throws', () => {
    const repo = new InMemoryGameTypeRepository()
    const useCase = new ArchiveGameTypeUseCase(repo)

    expect(() => useCase.invoke('nonexistent')).toThrow(NotFoundError)
  })
})
